src/tsickle.ts

```diff
@@ -692,18 +709,42 @@ class Annotator extends ClosureRewriter {
         continue;
       }
       this.generatedExports.add(name);
-      reexports.add(name);
+      reexports.add(sym);
     }
-
     return toArray(reexports.keys());
   }
 
   /**
+   * Write an `exports.` assignment for each type alias exported in the given `exports`.
+   * TypeScript by itself does not export non-value symbols (e.g. interfaces, typedefs), as it
+   * expects to remove those entirely for runtime. For Closure, types must be
+   * exported as downstream code will import the type.
+   *
+   * The tsickle pass turns interfaces into values by generating a `function MyInterface() {}` for
+   * them, so in the second conversion pass, TypeScript does export a value for them. However for
+   * pure typedefs, tsickle only generates a property access with a JSDoc comment, so they need to
+   * be exported explicitly here.
+   */
+  private emitTypeDefExports(exports: ts.Symbol[]) {
+    if (this.options.untyped) return;
+    const typeChecker = this.program.getTypeChecker();
+    for (let sym of exports) {
+      if (sym.flags & ts.SymbolFlags.Alias) sym = typeChecker.getAliasedSymbol(sym);
+      const isTypeAlias =
+          sym.flags & ts.SymbolFlags.TypeAlias && (sym.flags & ts.SymbolFlags.Value) === 0;
```

Comment:
Can you add some more parens here?  I have been bitten so hard by the C precedence bug around & vs == that I am terrified of this code, even if tests do show it works :)